lib/manager/poetry/artifacts.js

```diff
@@ -0,0 +1,76 @@
+const upath = require('upath');
+const process = require('process');
+const fs = require('fs-extra');
+const { exec } = require('child-process-promise');
+
+module.exports = {
+  getArtifacts,
+};
+
+async function getArtifacts(
+  packageFileName,
+  updatedDeps,
+  newPackageFileContent,
+  config
+) {
+  await logger.debug(`poetry.getArtifacts(${packageFileName})`);
+  if (updatedDeps === undefined || updatedDeps.length < 1) {
+    logger.debug('No updated poetry deps - returning null');
+    return null;
+  }
+  const lockFileName = 'poetry.lock';
+  const existingLockFileContent = await platform.getFile(lockFileName);
+  if (!existingLockFileContent) {
+    logger.debug(`No ${lockFileName} found`);
+    return null;
+  }
+  const localPackageFileName = upath.join(config.localDir, packageFileName);
+  const localLockFileName = upath.join(config.localDir, lockFileName);
+  let stdout;
+  let stderr;
+  const startTime = process.hrtime();
+  try {
+    await fs.outputFile(localPackageFileName, newPackageFileContent);
+    logger.debug(`Updating ${lockFileName}`);
+    const cwd = config.localDir;
+    for (let i = 0; i < updatedDeps.length; i += 1) {
+      const dep = updatedDeps[i];
+      const cmd = `poetry update --lock --no-interaction ${dep}`;
+      ({ stdout, stderr } = await exec(cmd, {
```

Comment:
We need to prepare this for Docker too, like existing managers. I can create the `renovate/poetry` image soon.

Reply:
I have adapted `getArtifacts` for docker `binarySource` (using generic code from `cargo` manager's `getArtifacts`).